include/rapidjson/document.h

```diff
@@ -952,12 +956,16 @@ class GenericValue {
         if (IsUint64()) {
             uint64_t u = GetUint64();
             volatile double d = static_cast<double>(u);
-            return static_cast<uint64_t>(d) == u;
+            return (d >= 0.0)
+                && (d < (static_cast<double>(std::numeric_limits<uint64_t>::max()/2 + 1)) * 2)
```

Comment:
Casting a double that's out-of-range for the uint64_t is undefined behaviour, so this equality test has undefined results. I'm not 100% clear on what exactly IsLosslessDouble promises if it returns true, but it's certainly possible that you could get true from the equality test when you expect false.